Assets/_BForBoss/_Leaderboard/Scripts/Username/ForcedSetUsernameViewBehaviour.cs

```diff
@@ -0,0 +1,38 @@
+using PerigonGames;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class ForcedSetUsernameViewBehaviour : SetUsernameViewBehaviour
+    {
+        public override void Initialize(SetUsernameViewModel viewModel = null)
+        {
+            base.Initialize(_viewModel);
+            if (_viewModel.IsUsernameAlreadySet())
+            {
+                HidePanel();
+            }
+            else
+            {
+                ShowPanel();
+            }
+        }
+        
+        protected override void BindViewModel()
+        {
+            base.BindViewModel();
+            _viewModel.OnSuccess += HidePanel;
```

Comment:
Should `_viewModel.RemoveSubscribers();` get called somewhere in this class? Or is this being handled elsewhere?

Reply:
The parent class already calls it on destroy